### ISSUE TITLE
Crash when integrating the xcframework that contains mixpanel

### ISSUE THREAD

**Issue by uzair** [mixpanel/mixpanel-swift#474]
I have create my custom framework and integrated mixpanel in it. When i create xcframework and integrate it in my sample project i'm getting the below crash: 

dyld: lazy symbol binding failed: Symbol not found: _$s8MixpanelAAC10initialize5token13launchOptions13flushInterval12instanceName21automaticPushTracking06optOutL9ByDefaultAA0A8InstanceCSS_SDySo019UIApplicationLaunchE3KeyaypGSgSdSSS2btFZTj Referenced from: /Users/uzairshahzad/Library/Developer/CoreSimulator/Devices/F078F0D6-1469-430B-AB7B-35DDAECCC333/data/Containers/Bundle/Application/71E16C58-078E-4A8F-AF48-3ADAE35A210F/MixpanelExample.app/Frameworks/TGPaySDK.framework/TGPaySDK Expected in: /Users/uzairshahzad/Library/Developer/CoreSimulator/Devices/F078F0D6-1469-430B-AB7B-35DDAECCC333/data/Containers/Bundle/Application/71E16C58-078E-4A8F-AF48-3ADAE35A210F/MixpanelExample.app/Frameworks/Mixpanel.framework/Mixpanel

**Comment by zihejia** [comment]
hi @uzair , could you share the integration code related to Mixpanel? Also, how do you integrate Mixpanel? For right now, we suggest using SPM.

**Comment by zihejia** [comment]
hi @uzair , I will close this issue for now. Please feel free to reopen the issue anytime you get a chance to provide more information and the problem still exists.

**Comment by jvigneshcs** [comment]
Even myself facing the similar issue. We are using Cocoapods for integration

**Comment by jvigneshcs** [comment]
> hi @jvigneshcs , which Mixpanel version are you using?

I tried with 4.1.2 (latest)

**Comment by jaredmixpanel** [comment]
@jvigneshcs When I try to execute `./GenerateXCFramework.sh` from the terminal, I get...

```
2023-07-19 10:31:33.525 xcodebuild[82369:13621943] DVTCoreDeviceEnabledState: DVTCoreDeviceEnabledState_Disabled set via user default (DVTEnableCoreDevice=disabled)
** ARCHIVE FAILED **
```

Have you experimented with using the `:linkage` option in your Podfiles?

`use_frameworks! :linkage => :static` and `use_frameworks! :linkage => :dynamic`

See:

https://medium.com/microsoft-mobile-engineering/mixing-static-and-dynamic-linking-in-cocoapods-83b6d4252c59

https://guides.cocoapods.org/syntax/podfile.html#use_frameworks_bang

**Comment by jvigneshcs** [comment]
@jaredmixpanel , 

> Have you experimented with using the `:linkage` option in your Podfiles?
> 
> `use_frameworks! :linkage => :static` and `use_frameworks! :linkage => :dynamic`

I tried your suggestion with `:linkage`,
used `use_frameworks! :linkage => :dynamic` from MySDK framework project's pod file,
when I tried **without** `:linkage => :static` from MyApp application project's pod file, I got
> missing symbol called

when  I tried **with** `:linkage => :static` from MyApp application project's pod file, I got
> Library not loaded: @rpath/Mixpanel.framework/Mixpanel

**Comment by jaredmixpanel** [comment]
Nevermind, got it to work. I had to remove `-sdk` flag from your `xcodebuild archive` commands in your build script to get the `xcodebuild -create-xcframework` step to work.

**Comment by jaredmixpanel** [comment]
@jvigneshcs I've experimented with different build configurations. I'm able to get it building and running without crashing but I do see messages like this in the console:

`objc[95678]: Class _TtC8Mixpanel15AutomaticEvents is implemented in both .../Xcode/DerivedData/MyApp-edscjgdqpchjdxfddizvrnvjpylm/Build/Products/Release-iphonesimulator/MySDK.framework/MySDK (0x103132a58) and .../CoreSimulator/Devices/89B0526D-C307-4DDA-8BF3-E794CAF74C03/data/Containers/Bundle/Application/60BD3F4D-0014-44F7-82EA-02ACCF9764E1/MyApp.app/MyApp (0x102f87848). One of the two will be used. Which one is undefined.`

In MySDK - MyAPI.swfit it seems like you will need to `@_implementationOnly import Mixpanel`. If I do that, then I can successfully run MyApp with the above warnings in the console if use `:linkage => :static` in MySDK's Podifile. It also seems to work if I remove CocoaPods altogether and use SPM from both MySDK and MyApp (again with the `implemented in both` warnings).

So I do not think this an issue with the Mixpanel SDK. I think this use case is running up against limitations in Swift.

Some further reading:
https://forums.swift.org/t/exported-and-fixing-import-visibility/9415
https://forums.swift.org/t/update-on-implementation-only-imports/26996

**Comment by jaredmixpanel** [comment]
@jvigneshcs One workaround I did find is to use the [Mixpanel Objective-C SDK ](https://github.com/mixpanel/mixpanel-iphone) from MySDK instead of the Swift SDK. If however, MyApp also uses the Objective-C SDK then we're back to the `implemented in both` console warnings.

**Comment by jvigneshcs** [comment]
> @jvigneshcs One workaround I did find is to use the [Mixpanel Objective-C SDK ](https://github.com/mixpanel/mixpanel-iphone) from MySDK instead of the Swift SDK. If however, MyApp also uses the Objective-C SDK then we're back to the `implemented in both` console warnings.

@jaredmixpanel yeah we have already tried this approach and we didn't face the crash issue which is happening with mixpanel-swift pod.
Let me also try your other suggestions as well.
